### DISCUSSION TITLE
Remove arbitary

### DISCUSSION THREAD

**Discussion by prrobert-code** [ViennaTools/ViennaPS#96] [Q&A]
Hello, thank you for providing this simulation tool! I have started to simulate a 3D domain and would like to simulate the following process sequence: mask a certain area - deposit material - flatten structure - remove mask to start etching process. However, I am unable to remove the mask as I can only remove the last layer added (geometry.removeTopLevelSet()). Is there any functionality to remove an arbitrary level set or any other procedure to remove a mask within a process simulation? Thanks for any advice / help!

**Comment by tobre1** [comment]
Hi,

Due to the specific layer wrapping scheme, it's not that straightforward to remove an arbitrary level set from the domain. Whenever you add a new level set, it wraps all the lower level sets, making it difficult to remove a non-top layer directly. To remove a level set that's not at the top, you'd need to perform boolean operations with the other level sets. However, this approach can sometimes lead to numerical issues, especially if some layers are identical in certain regions.

We’re addressing this limitation in the upcoming version 3.2.0, which will include a new functionality to help remove arbitrary level sets and materials from the domain. Please stay tuned for the 3.2.0 release.

Thank you for your patience!

**Reply by prrobert-code** [reply]
Dear maintainer, thanks for your time and answer! I am looking forward to the new release. 
In the meantime, I have indeed been able to remove a non-top layer using boolean operations as you suggested. I first created a separate layer of the material deposited on the substrate + mask structure. With this separate deposited material I reconstructed the structure by combining the substrate with the deposited material to get rid of the mask layer. The direct removal of the mask/resist layer would complement your tool very well.
